hathor/verification/verification_dependencies.py

```diff
@@ -0,0 +1,68 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from dataclasses import dataclass
+
+from typing_extensions import Self
+
+from hathor.daa import DifficultyAdjustmentAlgorithm
+from hathor.feature_activation.feature_service import BlockSignalingState, FeatureService
+from hathor.transaction import Block
+from hathor.transaction.storage.simple_memory_storage import SimpleMemoryStorage
+from hathor.transaction.transaction import Transaction
+
+
+@dataclass(frozen=True, slots=True)
+class VertexDependencies:
+    """A dataclass of dependencies necessary for vertex verification."""
+    storage: SimpleMemoryStorage
+
+
+@dataclass(frozen=True, slots=True)
+class BlockDependencies(VertexDependencies):
+    """A dataclass of dependencies necessary for block verification."""
+    signaling_state: BlockSignalingState
+
+    @classmethod
+    def create(cls, block: Block, daa: DifficultyAdjustmentAlgorithm, feature_service: FeatureService) -> Self:
+        """Create a block dependencies instance."""
+        assert block.storage is not None
+        signaling_state = feature_service.is_signaling_mandatory_features(block)
```

Comment:
Why is it part of the dependencies?

Reply:
Because since the Feature Activation process is dynamic, it is not straightforward to determine which blocks are necessary to calculate this state in advance (differently from the DAA, for example, where it's easy to determine the number of necessary parent blocks).

In order to determine the necessary blocks, we would have to run the algorithm anyway, so there's almost no difference in cost between determining all block dependencies in advance, or determining the state itself in advance.

I suggest we implement it like this for now, and then after the verification is moved to another process, we measure the cost of doing this calculation in main process, to decide if it's worth the refactor (we would need to change Feature Activation code in order to get just the required blocks).

Reply:
As discussed in our call, we'll keep this for now. After completing the Multiprocess Verification, we'll measure its performance and determine if it should be improved or not.